xmodule/capa/capa_problem.py

```diff
@@ -495,6 +509,51 @@ def get_grade_from_current_answers(self, student_answers):
 
         return newcmap
 
+    def get_grade_from_answers(
+        self, student_answers: Optional[dict], correct_map: Optional[CorrectMap] = None
+    ) -> CorrectMap:
+        """
+        This method is based on `get_grade_from_current_answers` but it is used when
+        `ENABLE_GRADING_METHOD_IN_PROBLEMS` feature flag is enabled.
+
+        This method optionally receives a `correct_map` to be used instead
+        of the current one. This allows to make a rescore using each of the
+        correct maps stored in the `correct_map_history`.
+
+        Args:
+            student_answers (Optional[dict]): A dictionary with the student answers.
+            correct_map (CorrectMap, optional): A CorrectMap object with the correct answers.
+                Defaults to None.
+
+        Returns:
+            CorrectMap: A CorrectMap object with the results of the grading.
+        """
+        if not correct_map:
+            correct_map = self.correct_map
+
+        newcmap = CorrectMap()
+        # Call each responsetype instance to do actual grading
+        for responder in self.responders.values():
+            # File objects are passed only if responsetype explicitly allows
+            # for file submissions.  But we have no way of knowing if
+            # student_answers contains a proper answer or the filename of
+            # an earlier submission, so for now skip these entirely.
+            # TODO: figure out where to get file submissions when rescoring.
+            if 'filesubmission' in responder.allowed_inputfields:
```

Comment:
I may be missing something, but it seems that if `filesubmission` is an allowed input field, we should not allow flexibility in grading policy, (i.e., we should then block requests to change grading policy away from the current first answer policy). Relaxing this constraint can then become a future feature, possibly.

Reply:
For this case, we could update the message in the **Grading Method** field, stating that it is not recommended to change the grading method when files are accepted. What do you think?

Reply:
For me, that's a good middle ground.